torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -29,8 +29,14 @@ bool idIsALeafDomain(IterDomain* id, TensorView* tv) {
 
 } // namespace
 
-IterDomainGraph::IterDomainGraph(Fusion* fusion) {
+IterDomainGraph::IterDomainGraph(Fusion* fusion, bool allow_self_mapping) {
   build(fusion);
+
+  if (!allow_self_mapping) {
+    TORCH_CHECK(
```

Comment:
Can you expand the error reporting with information about the domains causing the failure?

Reply:
+1
Could collect this information in `selfMappingExists`, and print it on failure.

Reply:
Error message added